src/com/hms/flexyosisoftconnector/OSIsoftServer.java

```diff
@@ -210,13 +235,42 @@ public int setTagWebId(Tag tag) throws JSONException {
    }
 
    // Initializes a list of tags
-   public int initTags(ArrayList tagList) throws JSONException{
-      int retval = NO_ERROR;
-      for (int i = 0; i < tagList.size(); i++) {
-         int res = setTagWebId((Tag) tagList.get(i));
-         if (res != NO_ERROR) return retval = res;
-      }
-      return retval;
+    public int initTags(String serverIp, ArrayList tagList, int communicationType) throws JSONException{
+        int retval = NO_ERROR;
+        int res;
+
+        switch(communicationType) {
+           case OSIsoftConfig.omf:
+              //OMF setup
+
+              // setup type
+              String responseFilename = "/usr/response.txt";
+              String messageTpyeHeader = "&messagetype=type";
+              setTypeBody();
+              res = RequestHTTPS(omfUrl, "Post", omfPostHeaders + messageTpyeHeader, batchBuffer.toString(), "", responseFilename);
+
+              // setup containers
+              messageTpyeHeader = "&messagetype=container";
+
+              // set batch buffer with list of tag containers to initialize for if they have not been already
+              setContainerJson(tagList);
+              res = RequestHTTPS(omfUrl, "Post", omfPostHeaders + messageTpyeHeader, batchBuffer.toString(), "", responseFilename);
+              if (res != NO_ERROR)
+                 retval = res;
+              break;
+           case OSIsoftConfig.piwebapi:
+              // old PIWEBAPI setup
+              for (int i = 0; i < tagList.size(); i++) {
+                 res = setTagWebId((Tag) tagList.get(i));
+                 if (res != NO_ERROR) return retval = res;
+              }
+              break;
+           default:
+              Logger.LOG_ERR("Immpossible to reach here, checks done in config class.");

```

Comment:
Spelling : Impossible 

Also this probably relates to the same comment as before about printing a message that makes sense considering you are not printing a line number.